### ISSUE TITLE
More specific error

### ISSUE THREAD

**Issue by careri** [Starcounter/Starcounter#2629]
I get this weaver error in VS:
An unspecified error caused the operation to fail. Illegal element for the SynonymousTo attribute (ScErrUnspecified (SCERR999))

It would be helpful if it pointed out ther erroneous class

The administrator log:
 HelpLink=https://github.com/Starcounter/Starcounter/wiki/SCERR999
 
2015-03-25 08:38:36	Error	sc://headsutv180/personal	Starcounter.Server	ScErrWeavingError (SCERR2030): At least one error has been detected during weaving of the application code. Process exit code: 999.
 Arguments=--maxerrors=1 --ErrorParcelId=A4A7D6FA-EB34-442A-B579-DBB1DBB859E3 Weave "d:\source\Scdb2Apps\HeadsAdmin\WebApp\App\bin\Debug\HADWebApp.exe" --outdir="C:\Users\careri\Documents\Starcounter\Personal\Temp\Default\Default-20150325T083823986\weaver\0FC2D1E90BAB5AAB739E63DA7B3194A3062F424F"
 Output:
 A4A7D6FA-EB34-442A-B579-DBB1DBB859E3ScErrUnspecified (SCERR999): An unspecified error caused the operation to fail. Illegal element for the SynonymousTo attribute
 Version: 2.0.3077.3.
 Help page: https://github.com/Starcounter/Starcounter/wi

**Comment by per-samuelsson** [comment]
@careri 

Thanks for reporting!

I will assign it to me and eventually fix it.

**Comment by careri** [comment]
Is it possible to figure out what's generating the error by searching the weaver temp dir?

I can't quite figure out the problem by browsing the code.

**Comment by careri** [comment]
Can I attach a debugger to a process (which exe?) and catch the exception?

**Comment by per-samuelsson** [comment]
> Can I attach a debugger to a process (which exe?) and catch the exception?

You can run the weaver standalone, yes. But I'm not sure you will find anything, since you have a release build.

The weaver is in ```scweaver.exe```.

```
mkdir Test
cd Test
%StarcounterBin%\scweaver.exe "d:\source\Scdb2Apps\HeadsAdmin\WebApp\App\bin\Debug\HADWebApp.exe" 
```

There is an unofficial switch, ```--sc-debug```, that you can apply and it attach a debugger. But I'm unsure if it works in release builds. Otherwise, just run it via VS.

**Comment by careri** [comment]
Was just looking through the code, there is also the verbosity switch?
syntaxDefinition.DefineProperty("verbosity", "Sets the verbosity of the program (quiet, minimal, verbose, diagnostic). Minimal is the default.");

I'll try your suggestion

**Comment by per-samuelsson** [comment]
> Was just looking through the code, there is also the verbosity switch?

Yep, that might actually give you the reason. Try it. Apply the higest level, diagnostic.

```
scweaver --verbosity=diagnostic myapp.exe
```

**Comment by careri** [comment]
I did that, the last class processed. Is that the culprit?
ScAnalysis - Validating the database class DummyDB.DummyWorkstationBoundSoftware.
ScErrUnspecified (SCERR999): An unspecified error caused the operation to fail. Illegal element for the SynonymousTo attribute
Version: 2.0.3077.3.
Help page: https://github.com/Starcounter/Starcounter/wiki/SCERR999.

**Comment by careri** [comment]
Strange haven't changed that class since 4 days ago. I'll try deleting the temp folder

**Comment by per-samuelsson** [comment]
> I did that, the last class processed. Is that the culprit?

Hmm, dont exactly understand what you mean I'm afraid?

If you cant see anything particular with your synonym declarations, and want more hands-on help, maybe I can run your code and debug it.

**Comment by careri** [comment]
The last class written to the cmd log before raising the error:
ScAnalysis - Validating the database class DummyDB.DummyWorkstationBoundSoftware.

Is that the class that has a problematic SynonymousTo?

**Comment by careri** [comment]
I'll try attaching the debugger

**Comment by per-samuelsson** [comment]
> Is that the class that has a problematic SynonymousTo?

Yep.

**Comment by per-samuelsson** [comment]
> Strange haven't changed that class since 4 days ago.

Maybe you have changed any class being referenced from the ```DummyWorkstationBoundSoftware```? This can also affect the referencing one.

**Comment by careri** [comment]
We'll debugging didn't help much since the exception is thrown in PostSharp from this line:
postSharpObject.ProcessAssemblies(strategies);

I had added a new database class, but I forgot the ````[Database]```` attribute on the new class. Not quite sure why it crashed on analyzing ````DummyWorkstationBoundSoftware````. It had no direct reference to the new class. It has a SynonymousTo typed as ````DummyWorkstation````. But looking at the scweaver log, DummyWorkstation had already passed analyzing.

So I was using SynonymousTo on a class without the database-attribute.

**Comment by per-samuelsson** [comment]
> So I was using SynonymousTo on a class without the database-attribute.

OK. And this is not supported. This shows the importance of proper, informative error message. Really sorry for the bad experience. I will try fixing this as soon as I get the chance to.

**Comment by careri** [comment]
Always expect the user, at least me, to use the product as it wasn't intended ;)
Thanks for the quick responses.

**Comment by per-samuelsson** [comment]
Fixed as part of #3677.
